File: programming_in_c/Assignment-6/Q07.c
/*Q07. Write a C program to count total number of words in a string.*/

#include <stdio.h>
//finds length of a string without NULL character
int strLen (char len[]){
    int i = 0;
    while (len[i] != '\0'){
        i++;
    }
    return i - 1;
} 

void wordCount (char arr[]){
    int strlen = strLen (arr);
    int c = 0;
    for (int i = 0; i <= strlen; i++){
        if (arr[i] == ' ' || arr[i] == '\t')
            c++;
    }
    printf("Total Number of Words: %d\n", c+1);
}

int main(){
    char str[100];
    printf("Enter a String: ");
    fgets(str, sizeof(str), stdin);
    wordCount(str);
    return 0;
}

File: Java/codes/Q07.java
class Q07 {

  public static void main(String[] args) {
    int arr[] = { 10, 31, 45, 5, 3, 11 };
    int sum = 0;

    System.out.print("Array is: ");
    for (int j = 0; j < arr.length; j++) {
      System.out.print(" " + arr[j]);
    }
    System.out.print("\n");
    for (int i = 0; i < arr.length; i++) {
      sum = sum + arr[i];
    }
    System.out.print("Sum of Arrya is: " + sum);
  }
}
